OpenRA.Mods.Common/Traits/World/Locomotor.cs

```diff
@@ -288,31 +290,37 @@ bool CanMoveFreelyInto(Actor actor, Actor ignoreActor, CPos cell, CellConditions
 			if (cellBlocking == CellBlocking.Empty)
 				return true;
 
+			if (sharesCell && cellBlocking.HasCellBlocking(CellBlocking.FreeSubCell))
+				return true;
+
 			// We are blocked
-			if (ignoreActor == null && blocking.Overlaps(actor.Owner.PlayerMask))
+			if (ignoreActor == null && blocking.Overlaps(actor.Owner.PlayerMask) && cellBlocking < CellBlocking.Crushable)
 				return false;
 
-			if (cellBlocking.HasCellBlocking(CellBlocking.Crushable))
-				return true;
+			if (check == CellConditions.BlockedByMovers && cellBlocking < CellBlocking.Crushable)
+				return false;
 
-			if (sharesCell && cellBlocking.HasCellBlocking(CellBlocking.FreeSubCell))
+			// We can only do an early exit here when we know the unit is not moving.
+			// Because we cant distinguish between crushable and ally moving (which we need to fall back to the old code)
```

Comment:
IIRC there was some discussion earlier about inverting and replacing `IsMoving` with something like `StationaryBlocker`. IMO this would make the logic clearer for cells with a mix of stationary and moving units. 

`StationaryBlocker` retains its meaning if another unit in the cell is moving, whereas the current code will (i think?) set `IsMoving` if the cell has one moving unit, but clear it if there is one moving and one stationary unit.